doc/ref/configuration/minion.rst

```diff
@@ -635,6 +635,38 @@ FQDN (for instance, Solaris).
 
     append_domain: foo.org
 
+.. conf_minion:: minion_id_remove_domain
+
+``minion_id_remove_domain``
+---------------------
+
+.. versionadded:: 18.3.?
+
+Default: ``''``
+
+Remove a domain when the minion id is generated as a fully qualified domain
+name (either by the user provided ``id_function``, or by Salt). This is useful
+when the minions shall be named like hostnames. Can be a single domain (to
+prevent name clashes), or True, to remove all domains.
+
+Examples:
+ - minion_id_remove_domain = foo.org
+   - FQDN = king_bob.foo.org --> minion_id = king_bob
+   - FQDN = king_bob.bar.org --> minion_id = king_bob.bar.org
+ - minion_id_remove_domain = True
+   - FQDN = king_bob.foo.org --> minion_id = king_bob
+   - FQDN = king_bob.bar.org --> minion_id = king_bob
```

Comment:
Maybe rephrase this by just plain English, something like:

When `minion_id_remove_domain` is set to `foo.org`, then only this domain will be removed from the minion ID. For example, `king_bob.foo.org` will be `king_bob` and `king_bob.bar.org` will remain unchanged. In case `minion_id_remove_domain` is set to `True`, then all domains will be removed. For example, both `king_bob.foo.org` and `king_bob.bar.org` will be `king_bob`.

How's that sounds like? Feel free to adapt it and change it.

Reply:
I prefer table-list like schematic examples, also here.